Test/UpdateUserView.swift

```diff
@@ -9,8 +9,12 @@ import ComposableArchitecture
 import SwiftUI
 
 struct UpdateUserState: Equatable {
+  init(user: User?) {
+    self.user = user
+    self.username = user?.username ?? ""
+  }
 	var user: User?
-	var username = ""
+  var username: String
```

Comment:
So here I assume you are forcing to pass the "placeholder", in this case getting it from the actual user, instead of reading it from the "user.username" and setting it in onAppear. Maybe in the end is a matter of preference 🤔 , but if I choose this route I would also change the `state.username` here to `state.placeholder` ?